Tutorial from Manon but cute (#33)
scilus/nf-neuro-documentation: src/content/docs/pipeline/tutorial_step_5.mdx

   You should see this output:
    ```bash

                                            ,--./,-.
            ___     __   __   __   ___     /,-._.--~\
        |\ | |__  __ /  ` /  \ |__) |__         }  {
        | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                            `._,._,'

        nf-core/tools version 2.14.1 - https://nf-co.re
        There is a new version of nf-core/tools available! (3.2.0)

    INFO     Installing 'preproc_t1'                                                                                                                                                             
    INFO     Use the following statement to include this subworkflow:                                                                                                                            
                                                                                                                                                                                                
    include { PREPROC_T1 } from '../subworkflows/nf-neuro/preproc_t1/main' 
    ```

   </span>
   </CommandOutputs>

### 2. Add new data in the input pipeline structure

> To include a T1w image in the pipeline, follow the approach from [**Step 2**](/nf-neuro/pipeline/tutorial_step_2) for adding DWI images.
> The necessary steps are: 

Reviewer: ```suggestion
To include a T1w image in the pipeline, we will follow the same approach from [**Step 2**](/nf-neuro/pipeline/tutorial_step_2) as we did for the DWI images.
The required steps are: 
```

    <Steps>

>     1. Add T1w image path to `get_data` input structure.
> 
>     2. `emit` a new output channel called `anat`.
> 
>     3. Include the T1 channel data in the worflow and visualize it.
> 
>     4. Run the pipeline using `nextflow run main.nf --input data -profile docker`.

Reviewer: ```suggestion
    1. Add T1w image path to the `get_data` input structure.

    2. `emit` a new output channel called `anat`.

    3. Include the T1w channel data in the workflow and visualize it.

    4. Run the pipeline using `nextflow run main.nf --input data -profile docker`.
```

    </Steps>

    <Code code={main} lang="bash" title="Expected main.nf"/>
    <RunIcon class="inline-icon text-blue-300" aria-hidden /> **Now, you can run nextflow..**

   <CommandOutputs>
   <span slot="command">
    ```bash
    nextflow run main.nf --input data -profile docker
    ```
   </span>
   <span slot="output">
    ```bash
    Launching `main.nf` [evil_noether] DSL2 - revision: f131ccc34c

    executor >  local (2)
    [c8/fa8ee7] process > DENOISING_MPPCA (sub-003_ses-01)    [100%] 1 of 1 ✔
    [e9/6b32bf] process > RECONST_DTIMETRICS (sub-003_ses-01) [100%] 1 of 1 ✔

    [[id:sub-003_ses-01], /workspaces/nf-neuro-tutorial_test/data/sub-003/ses-01/anat/sub-003_ses-01_T1w.nii.gz]
    ```
   </span>
   </CommandOutputs>

> ### 3. Use the subworkflow and prepare the input subworkflow structure

Reviewer: ```suggestion
### 3. Prepare the input structure for the subworkflow and include it in your `main.nf`
```

#### a. Use the subworkflow as follow in your `main.nf`:

    ```nextflow
    PREPROC_T1( input_channel(s) )
    ```

#### b. Prepare structure input for the subworkflow

    As with modules, the API documentation also lists [**subworkflows**](/nf-neuro/api/categories), 
    providing information on **Inputs**, **Outputs** and the module list used in the 
    **Components** section:  [PREPROC_T1](/nf-neuro/api/subworflows/preproc_t1/).

>     The PREPROC_T1 process has 7 input channels defined, with only ch_image being mandatory.
>     Just like using an empty list for modules, using an **empty channel** allows the 
>     process to run without optional inputs.

Reviewer: ```suggestion
    The `PREPROC_T1` subworkflow has 7 input channels defined, with only `ch_image` being mandatory.
    Just like using an empty list for modules, using an **empty channel** allows the 
    process to run without supplying all optional inputs.
```

    <html>

>     <li>ch_image (Mandatory)</li>
>     <li>ch_template (Optional)</li>
>     <li>ch_probability_map (Optional)</li>
>     <li>ch_mask_nlmeans (Optional)</li>
>     <li>ch_ref_n4 (Optional)</li>
>     <li>ch_ref_resample (Optional)</li>
>     <li>ch_weights (Optional)</li>

Reviewer: ```suggestion
    <li>`ch_image` (Mandatory)</li>
    <li>`ch_template` (Optional)</li>
    <li>`ch_probability_map` (Optional)</li>
    <li>`ch_mask_nlmeans` (Optional)</li>
    <li>`ch_ref_n4` (Optional)</li>
    <li>`ch_ref_resample` (Optional)</li>
    <li>`ch_weights` (Optional)</li>
```

    </html>

>     Use the `Channel.empty()` function for optional inputs, try to define the input structure for PREPROC_T1.

Reviewer: ```suggestion
    Using the `Channel.empty()` function for optional inputs, you can define the input structure for `PREPROC_T1` using the newly fetched `inputs.anat` channel.
```

    ```nextflow

        PREPROC_T1(
            inputs.anat,
            Channel.empty(),
            Channel.empty(),
            Channel.empty(),
            Channel.empty(),
            Channel.empty(),
            Channel.empty()
        )
    ```

### 4. Configure your subworkflow

> Configuring a subworkflow requires understanding the modules used and the associated parameters.
> 
> The [PREPROC_T1](https://scilus.github.io/nf-neuro/api/subworkflows/preproc_t1/) API documentation 
> lists the modules included in the subworkflow in the **Components** section : 

Reviewer: ```suggestion
Configuring a subworkflow requires understanding the modules used within the subworkflow and their associated parameters. To properly use them, we will need to configure each of the module similarly to what we did for `RECONST_DTIMETRICS`. 

The [`PREPROC_T1`](https://scilus.github.io/nf-neuro/api/subworkflows/preproc_t1/) API documentation 
lists the modules included in the subworkflow in the **Components** section : 
```

    <html>
        <li>denoising/nlmeans</li>
        <li>preproc/n4</li>
        <li>image/resample</li>
        <li>betcrop/antsbet</li>
        <li>betcrop/synthbet</li>
        <li>image/cropvolume</li>
    </html>

However, it does not yet provide a list of the parameters associated with the subworkflows. 
You can find the parameters for the subworkflow in the `./subworkflow/nf-neuro/preproc_t1/main.nf` 
file or the `nextflow.config` file within the **tests** folder (`./subworkflow/nf-neuro/preproc_t1/tests/nextflow.config`).

> For simplicity's sake, we will enable the `denoising` and `synthbet` options (set to `true`) and 
> disable all other options (set to `false`). 

Reviewer: ```suggestion
For the purpose of this tutorial, we will only enable the `denoising` and `synthbet` options (set to `true`) and 
disable all other options (set to `false`).
```

Add the following to your `nextflow.config` file:

```nextflow
    // Add those line after params.output = 'result'

    // ** Subworkflow PREPROC T1 **
    params.preproc_t1_run_denoising = true
    params.preproc_t1_run_N4 = true
    params.preproc_t1_run_resampling = false
    params.preproc_t1_run_ants_bet = false
    params.preproc_t1_run_synthbet = true
    params.preproc_t1_run_crop = false

```

> Since we are disabling all options except for `denoising` and `synthbet`, we only need to 
> include the parameters for the modules corresponding to those options. 
> Note that the `denoising` module does not require any specific parameters, so it can be ignored in the configuration.

Reviewer: ```suggestion
Since we are disabling all options except for `denoising` and `synthbet`, we only need to 
include specific parameters for those two modules. 
Note that the `denoising` module does not require any specific parameters, so it can be ignored in the configuration.
```
